src/components/DataTable/DataTable.tsx

```diff
@@ -146,14 +150,21 @@ const DataTable = <T extends { id: number; selectDisabled?: boolean }>({
   }
 
   // sort, pagination, count
-  // 初回表示時は一番左側のsortableなcolumnを基準にソートする
-  const firstSortableColumn = columns.find(
-    (column) => column.sortable === true,
-  );
-  const [sortState, setSortState] = useOrderState<T>(
-    firstSortableColumn ? firstSortableColumn.selector : undefined,
-    firstSortableColumn ? firstSortableColumn.name : "",
+
+  // 初回表示時にdefaultSortFieldがなければ一番左側のsortableなcolumnを基準にソートする
+  const selectedColumn = columns.find(
+    (column) => column.name === defaultSortField,
   );
+  const firstSortableColumn = selectedColumn?.sortable
+    ? selectedColumn
+    : columns.find((column) => column.sortable === true);
+
+  const [sortState, setSortState] = useOrderState<T>({
+    isDesc: defaultSortOrder === "desc",
+    name: firstSortableColumn ? firstSortableColumn.name : "",
+    getValue: firstSortableColumn ? firstSortableColumn.selector : undefined,
+  });
```

Comment:
```suggestion
  const [sortState, setSortState] = useOrderState<T>({
    isDesc: defaultSortOrder === "desc",
    name: firstSortableColumn?.name || "",
    getValue: firstSortableColumn?.selector,
  });
```

書いてあることはこれと変わらなそうですね!!